inc/umock_c/umock_c_internal.h

```diff
@@ -15,14 +23,19 @@
 #include "umock_c/umockstring.h"
 #include "umock_c/umockautoignoreargs.h"
 
-#ifdef __cplusplus
-#include <cstdlib>
-extern "C" {
+#ifdef _MSC_VER
+#define UMOCK_C_WEAK
+#elif __GNUC__
+// GCC needs weak for the reals
```

Comment:
>GCC needs weak for the reals [](start = 3, length = 28)

Iiiiiiiiinteresting.... why would that be... I am curious!

Reply:
This is because gcc will try to find the real_ symbol regardless if it is or it is not used.
VS has obviously a different behavior.

Unfortunately I could not find a better way for this (yet). But there might always be some room for improvement

---
In reply to: [293559638](https://github.com/Azure/umock-c/pull/95#discussion_r293559638) [](ancestors = 293559638)